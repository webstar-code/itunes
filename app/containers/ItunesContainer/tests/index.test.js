/**
 *
 * Tests for ItunesContainer
 *
 */

import React from 'react';
import { timeout, renderProvider } from '@utils/testUtils';
import { fireEvent } from '@testing-library/dom';
import { ItunesContainerTest as ItunesContainer } from '../index';

describe('<ItunesContainer /> tests', () => {
  let submitSpy;

  beforeEach(() => {
    submitSpy = jest.fn();
  });
  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(<ItunesContainer dispatchGetItunes={submitSpy} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should call dispatchClearItunes on empty change', async () => {
    const getItunesSpy = jest.fn();
    const clearItunesSpy = jest.fn();
    const { getByTestId } = renderProvider(
      <ItunesContainer dispatchClearItunes={clearItunesSpy} dispatchGetItunes={getItunesSpy} />
    );
    fireEvent.change(getByTestId('search-bar'), {
      target: { value: 'a' }
    });
    await timeout(500);
    expect(getItunesSpy).toBeCalled();
    fireEvent.change(getByTestId('search-bar'), {
      target: { value: '' }
    });
    await timeout(500);
    expect(clearItunesSpy).toBeCalled();
  });

  it('should call dispatchGetItunes on change', async () => {
    const { getByTestId } = renderProvider(<ItunesContainer dispatchGetItunes={submitSpy} />);
    fireEvent.change(getByTestId('search-bar'), {
      target: { value: 'some artist' }
    });
    await timeout(500);
    expect(submitSpy).toBeCalled();
  });
});
