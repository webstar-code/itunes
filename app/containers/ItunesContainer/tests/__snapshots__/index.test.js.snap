// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItunesContainer /> tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="styles__Container-sc-1oy2h6v-0 NBLvG"
    >
      <div
        class="styles__RightContent-sc-1oy2h6v-3 hXNjsL"
      >
        <button
          aria-checked="false"
          class="ant-switch"
          role="switch"
          type="button"
        >
          <div
            class="ant-switch-handle"
          />
          <span
            class="ant-switch-inner"
          >
            light
          </span>
        </button>
      </div>
      <span
        class="ant-input-group-wrapper ant-input-group-wrapper-lg ant-input-search ant-input-search-large ant-input-search-with-button"
        style="width: 300px; margin-top: 20px;"
      >
        <span
          class="ant-input-wrapper ant-input-group"
        >
          <span
            class="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
          >
            <input
              class="ant-input ant-input-lg"
              data-testid="search-bar"
              placeholder="search artist"
              type="text"
              value=""
            />
            <span
              class="ant-input-suffix"
            >
              <span
                aria-label="close-circle"
                class="anticon anticon-close-circle ant-input-clear-icon-hidden ant-input-clear-icon"
                role="button"
                tabindex="-1"
              >
                <svg
                  aria-hidden="true"
                  data-icon="close-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="ant-input-group-addon"
          >
            <button
              class="ant-btn ant-btn-primary ant-btn-lg ant-input-search-button"
              type="button"
            >
              <span
                aria-label="search"
                class="anticon anticon-search"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="search"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                  />
                </svg>
              </span>
            </button>
          </span>
        </span>
      </span>
      <div
        class="ant-card ant-card-bordered ant-card-contain-tabs styles__CustomCard-sc-1oy2h6v-1 bDvfhk"
        style="margin: 20px; color: rgb(33, 37, 41);"
      >
        <div
          class="ant-card-head"
        >
          <div
            class="ant-card-head-wrapper"
          />
          <div
            class="ant-tabs ant-tabs-top ant-tabs-large ant-card-head-tabs"
          >
            <div
              class="ant-tabs-nav"
              role="tablist"
            >
              <div
                class="ant-tabs-nav-wrap"
              >
                <div
                  class="ant-tabs-nav-list"
                  style="transform: translate(0px, 0px);"
                >
                  <div
                    class="ant-tabs-tab ant-tabs-tab-active"
                  >
                    <div
                      aria-selected="true"
                      class="ant-tabs-tab-btn"
                      role="tab"
                      tabindex="0"
                    >
                      songs
                    </div>
                  </div>
                  <div
                    class="ant-tabs-tab"
                  >
                    <div
                      aria-selected="false"
                      class="ant-tabs-tab-btn"
                      role="tab"
                      tabindex="0"
                    >
                      videos
                    </div>
                  </div>
                  <div
                    class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
                  />
                </div>
              </div>
              <div
                class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
              >
                <button
                  aria-controls="null-more-popup"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-hidden="true"
                  class="ant-tabs-nav-more"
                  id="null-more"
                  style="visibility: hidden; order: 1;"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="ellipsis"
                    class="anticon anticon-ellipsis"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="ellipsis"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <div
              class="ant-tabs-content-holder"
            >
              <div
                class="ant-tabs-content ant-tabs-content-top"
              >
                <div
                  aria-hidden="false"
                  class="ant-tabs-tabpane ant-tabs-tabpane-active"
                  role="tabpanel"
                  tabindex="0"
                />
                <div
                  aria-hidden="true"
                  class="ant-tabs-tabpane"
                  role="tabpanel"
                  style="display: none;"
                  tabindex="-1"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
          style="margin: 0px; background: rgb(247, 247, 247);"
        >
          <div
            class="styles__Item-sc-1oy2h6v-2 cjavSI"
            style="margin: 0px; color: rgb(33, 37, 41);"
          >
            <div
              class="ant-card ant-card-bordered styles__CustomCard-sc-1wt29b6-0 cdteXX"
              color="false"
            >
              <div
                class="ant-card-body"
              >
                <p
                  class="T__StyledText-gjlic1-0 cbAvsY"
                  data-testid="t"
                >
                  Search for songs by entering the artist name in the search box
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
